### PR TITLE
Remove old Istio tests, add new Istio suite with regression values

### PR DESCRIPTION
**Pull request by npolshakova**
# Description
Cleans up old Istio regression tests. Adds regression test with old helm values to new framework. 

## API changes
None

## Code changes
- Removes old Istio regression tests from ci pipelines and code
- Removes old Istio glooctl tests 
- Removes Istio installation from setup script
- Adds a new Istio helm setup to the new framework that uses the deprecated values (enableIstioSidecarOnGateway) to check for regressions 
- Adds missing port routing test with edge apis. 

## CI changes
- Removed Istio regression tests 

## Docs changes
None

# Context

## Interesting decisions
None

## Testing steps
- Validated k8s and gloo edge Istio integration tests run locally

## Notes for reviewers
None

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/regression-tests.yaml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
         # upgrade tests are run on LTS but not on main branch, for main they are run nightly
         # ingress will be deprecated from 1.17. Ref: https://solo-io-corp.slack.com/archives/G01EERAK3KJ/p1716389614777799
         # this is the github action version of ternary op
-        kube-e2e-test-type: [ 'gateway', 'gloo', 'ingress', 'helm', 'gloomtls', 'glooctl', 'upgrade', 'istio' ]
```
Comment (sam-heilbron) [inline .github/workflows/regression-tests.yaml:52]: 🎉

File: test/kube2e/istio/README.md [hunk 2]
```diff
@@ -1,287 +0,0 @@
-An overview of some key concepts related to this test suite:
```
Comment (sam-heilbron) [inline test/kube2e/istio/README.md:1]: This seems like valuable information, that likely deserved a different place originally. Are we safe to remove this entirely, or should we be moving some/all of this to https://github.com/solo-io/gloo/tree/main/devel/architecture?

File: test/kubernetes/e2e/features/port_routing/edge_gateway_suite.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+Each port routing test:
+ 1. Attach HttpRoute with different port/targetport definition per test
+ 2. Remove HttpRoute, proxy still exists without any routes
+
+TearDownSuite:
+ 1. Deletes the k8s Gateway
+ 2. Proxy de-provisioned
+*/
+func NewEdgeGatewayApiTestingSuite(ctx context.Context, testInst *e2e.TestInstallation) suite.TestingSuite {
```
Comment (sam-heilbron) [inline test/kubernetes/e2e/features/port_routing/edge_gateway_suite.go:44]: nit: EdgeGateway implies those are the APIs we use, but the commentary for this fucntions is about k8s Gateway API

File: test/kubernetes/e2e/features/port_routing/testdata/edge-routing.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Don't set namespace, apply to test installation namespace
```
Comment (sam-heilbron) [inline test/kubernetes/e2e/features/port_routing/testdata/edge-routing.yaml:1]: ❤️

**Review by sam-heilbron — COMMENTED** [review]
looking great! Some small comments

File: test/kubernetes/e2e/features/port_routing/edge_gateway_suite.go (hunk 3 above)
Comment (npolshakova) [reply test/kubernetes/e2e/features/port_routing/edge_gateway_suite.go:44]: Good catch! I'll update the comment!

File: test/kube2e/istio/README.md (hunk 2 above)
Comment (npolshakova) [reply test/kube2e/istio/README.md:1]: Good idea, let's move it to the architecture docs!

Comment (npolshakova) [reply test/kube2e/istio/README.md:1]: Or do you think https://github.com/solo-io/gloo/blob/main/docs/content/guides/integrations/service_mesh/istio.md is better? The devel/architecture feels more developer facing, so I think that makes more sense, but a lot of similar information is in the `guides/integrations` docs.

Comment (sam-heilbron) [reply test/kube2e/istio/README.md]: Yeah I figure in general the most important part is that we have some docs. I tend to like the developer documentation because it means we can add a lot of context that is relevant to the code, which users might not care about. Plus, as we build our user-facing docs, the docs team can then leverage that.

**Review by sam-heilbron — APPROVED** [review]

**Review by jenshu — APPROVED** [review]
